pkg/agent/resourcemanager/fetcher/kubelet/topology/topology_adapter.go

```diff
@@ -315,6 +342,42 @@ func (p *topologyAdapterImpl) addNumaSocketChildrenZoneNodes(generator *util.Top
 	return nil
 }
 
+// addNumaSocketChildrenZoneNodes add the child nodes of socket or numa zone nodes to the generator, the child nodes are
+// generated by generateZoneNode according to TopologyLevel, Type and Name in TopologyAwareAllocatableQuantityList
+func (p *topologyAdapterImpl) addNICNumaChildrenZoneNodes(generator *util.TopologyZoneGenerator,
```

Comment:
We can make this function more generic by using a map configuration (key: device.ResourceName, value: ZoneType). If the user has set this map, the function can construct NUMA's child zones by iterating through allocatableResources.Devices. If device.ResourceName is present in the map configuration, the zone name will be the device ID.

Reply:
I have refactored this pr. Could you PTAL?